src/components/Dashboard.vue

```diff
@@ -243,19 +249,28 @@ export default Vue.extend({
       this.isOpeningRequirements = !this.isOpeningRequirements;
     },
 
-    convertSemesters(firebaseSems: readonly FirestoreSemester[]) {
-      const semesters = firebaseSems.map(firebaseSem => {
-        const appSemester = firestoreSemesterToAppSemester(
+    convertSemesters(
+      firebaseSems: readonly FirestoreSemester[],
+      subjectColors: { readonly [subject: string]: string },
```

Comment:
What's the purpose of the refactor where subjectColors is applied to convertSemesters? How i this better than the former solution

Reply:
Then convertSemesters is no longer entangled with `addColor`, which has some side effect (doing some firestore operations) that's hard to reason about.

Reply:
@hanywang2